packages/terra-folder-tree/src/FolderTree.jsx

```diff
@@ -54,6 +55,72 @@ const FolderTree = ({
   intl,
 }) => {
   const folderTreeID = `folder-tree-${uuidv4()}`;
+  const listNode = useRef();
+
+  const handleKeyDown = event => {
+    const listItems = listNode.current.querySelectorAll('[data-item-show-focus=true]');
+    // Remove all hidden list items
+    const visibleListItems = Array.prototype.slice.call(listItems).filter((item) => {
+      let parent = item.parentNode;
+      while (parent && parent !== listNode.current) {
+        if (parent.hasAttribute('hidden')) {
+          return false;
+        }
+        parent = parent.parentNode;
+      }
+      return true;
+    });
+    const currentIndex = Array.from(visibleListItems).indexOf(event.target);
+    const lastIndex = visibleListItems.length - 1;
+
+    const handleHomeKey = () => visibleListItems[0].focus();
+    const handleEndKey = () => visibleListItems[lastIndex].focus();
+
+    switch (event.nativeEvent.keyCode) {
+      case KeyCode.KEY_END:
```

Comment:
Do we need to prevent the default behavior for every key we support?  What undesired actions are we preventing?

Reply:
I'm not sure actually, but I believe it's pretty standard for custom navigation elements like this to have them. Let me run some tests to see if removing them is fine